sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/ConnectionStringTokenParser.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.RegularExpressions;
+
+namespace Azure.Messaging.EventHubs.Tests
+{
+    /// <summary>
+    ///   General purpose parser that allows to retrieve a token in connection string.
+    /// </summary>
+    ///
+    internal static class ConnectionStringTokenParser
+    {
+        /// <summary>It matches a pair "key=value" in a connection string</summary>
+        private static readonly Regex tokenValueRegex = new Regex("(?<tokenkey>[^;]+)=(?<tokenvalue>[^;]+)(;|$)", RegexOptions.IgnoreCase | RegexOptions.Compiled);
```

Comment:
nit:  The names for these should start with a capital (as they're intended as `const` semantics)